packages/mds-compliance-service/repository/index.ts

```diff
@@ -0,0 +1,215 @@
+        throw RepositoryError(`end_time of ${p_end_time} cannot be smaller than start_time of ${start_time}`)
+      }
+      try {
+        const connection = await connect('ro')
+        const query = connection
+          .getRepository(ComplianceSnapshotEntity)
+          .createQueryBuilder()
+          .where(`compliance_as_of >= ${start_time}`)
+          .andWhere(`compliance_as_of <= ${p_end_time}`)
+        if (isDefined(provider_ids)) {
+          query.andWhere('provider_id IN (:...provider_ids)', { provider_ids })
+        }
+        if (isDefined(policy_ids)) {
+          query.andWhere('policy_id IN (:...policy_ids)', { policy_ids })
+        }
+
+        query.orderBy('compliance_as_of')
+        const entities = await query.getMany()
+        return entities.map(ComplianceSnapshotEntityToDomain.mapper())
+      } catch (error) {
+        throw RepositoryError(error)
+      }
+    }
+    throw RepositoryError('start_time not provided')
+  }
+
+  public getComplianceSnapshotsByIDs = async (ids: UUID[]): Promise<ComplianceSnapshotDomainModel[]> => {
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      const entities = connection
+        .getRepository(ComplianceSnapshotEntity)
+        .createQueryBuilder()
+        .where('compliance_snapshot_id IN (:...ids)', { ids })
+        .getMany()
+      return (await entities).map(ComplianceSnapshotEntityToDomain.mapper())
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public createComplianceSnapshot = async (
+    complianceSnapshot: ComplianceSnapshotDomainModel
+  ): Promise<ComplianceSnapshotDomainModel> => {
+    const { connect } = this
+    try {
+      const connection = await connect('rw')
+      const {
+        raw: [entity]
+      }: InsertReturning<ComplianceSnapshotEntity> = await connection
+        .getRepository(ComplianceSnapshotEntity)
+        .createQueryBuilder()
+        .insert()
+        .values([ComplianceSnapshotDomainToEntityCreate.map(complianceSnapshot)])
+        .returning('*')
+        .execute()
+      return ComplianceSnapshotEntityToDomain.map(entity)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public createComplianceSnapshots = async (
+    ComplianceSnapshots: ComplianceSnapshotDomainModel[]
+  ): Promise<ComplianceSnapshotDomainModel[]> => {
+    const { connect } = this
+    try {
+      const connection = await connect('rw')
+      const { raw: entities }: InsertReturning<ComplianceSnapshotEntity> = await connection
+        .getRepository(ComplianceSnapshotEntity)
+        .createQueryBuilder()
+        .insert()
+        .values(ComplianceSnapshots.map(ComplianceSnapshotDomainToEntityCreate.mapper()))
+        .returning('*')
+        .execute()
+      return entities.map(ComplianceSnapshotEntityToDomain.map)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public deleteComplianceSnapshot = async (
```

Comment:
Do we want delete operations for Compliance Snapshots?

Reply:
I would think yes. Even if we don't expose a delete endpoint via the API, at some point we'll want to get rid of the older ones.

Reply:
Though now that I've thought about it, we probably don't want to delete a single `ComplianceSnapshot` at a time if we're cleaning house. As far as delete methods go, this one's not that useful. I'll remove it.